tests/basilisp/test_core_fns.lpy

```diff
@@ -1399,6 +1399,21 @@
   (is (= "lo w" (subs "hello world" 3 7)))
   (is (thrown? python/IndexError (subs "hello world" 12 3))))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; State Management Functions ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftest promises-test
+  (testing "timeout expires on time"
+    (let [p (promise)
+          start (time/perf-counter)
+          ;; block for 100ms
+          ret (deref p 100 :timed-out)
+          end (time/perf-counter)
+          diff (- end start)]
+      (is (= :timed-out ret))
+      (is (< 0.1 diff 0.2) (str "deref timeout is outside of [100ms, 200ms] range: " diff "ms")))))
```

Comment:
Is this going to be flaky?

Reply:
most likely, until we find a high enough tolerance value that is acceptable, I think. it already failed at a tolerance of 100ms which was borderline, I've increased the tolerance to 400ms. This should be give enough leeway, unless we think testing the upper-bound limit is always going to be problematic.